src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -61,27 +66,48 @@
     @Value("${ldap.enabled}")
     private boolean ldapEnabled;
 
+    @Value("${jasypt.encryptor.password:}")
+    private String jasyptEncryptorPassword;
+
     @Autowired
     private ISubscriptionRepository subscriptionRepository;
 
     @Override
-    public void addSubscription(Subscription subscription) throws JsonProcessingException, MongoWriteException {
+    public void addSubscription(Subscription subscription)
+            throws JsonProcessingException, MongoWriteException {
+        String authType = subscription.getAuthenticationType();
+        String username = subscription.getUserName();
+        String password = subscription.getPassword();
+        boolean authenticationDetailsProvided = isAuthenticationDetailsProvided(authType, username,
+                password);
+        if (authenticationDetailsProvided && !StringUtils.isEmpty(jasyptEncryptorPassword)
+                && !AuthenticationType.valueOf(authType).equals(AuthenticationType.NO_AUTH)) {
+            StandardPBEStringEncryptor encryptor = new StandardPBEStringEncryptor();
```

Comment:
Some duplications of code here, maybe the creation of encryptor could be created and prepared in a Factory?

Reply:
Ok so I changed name on EncryptionFormatter to Encryptor and simply use that for everything encryption. =P

Reply:
I had to split these again because of how the code in EI is written.